### ISSUE TITLE
[Feature request] Hide tree when Atom settings panel is open

### ISSUE THREAD

**Issue by nelson6e65** [atom/tree-view#1052]
### Description

Tree view stay open when I open settings, which uses some unnecessary space.

This is specially notable on low screen resolutions. 

### Steps to Reproduce

1. Open Atom Settings.

**Expected behavior:** Hide when settings is active.

**Actual behavior:** Keeping visible when settings are open.

**Reproduces how often:** 100%

### Versions

```
Atom    : 1.15.0
Electron: 1.3.13
Chrome  : 52.0.2743.82
Node    : 6.5.0
```

### Additional Information

![pantallazo-2017-03-27 19-58-11](https://cloud.githubusercontent.com/assets/9272498/24382989/efc0ce6c-1327-11e7-9b1a-f01437a1a5ff.png)

![pantallazo-2017-03-27 19-58-31](https://cloud.githubusercontent.com/assets/9272498/24382991/f5519d0c-1327-11e7-9c9e-ce00be1bd5fb.png)

**Comment by nelson6e65** [comment]
I achieve this by adding a listener in Atom init:

```coffee
# init.coffee
global.functions = require './functions'

atom.workspace.onDidChangeActivePaneItem (editor) ->
  functions.hideTreeIfNotEditor editor
```

```coffee
# functions.coffee

lastActiveWasEditor = false

exports.hideTreeIfNotEditor = (editor) ->
  view = atom.views.getView(editor)

  isEditor = editor.getPath?

  return if lastActiveWasEditor == isEditor

  lastActiveWasEditor = isEditor

  if isEditor
    atom.commands.dispatch(view, 'tree-view:show')
  else
    atom.commands.dispatch(view, 'tree-view:toggle')
```

But IDK if there is a better way?

> This should be a package feature 😞

**Comment by AnthonyWC** [comment]
Yes would definitely like to see a feature/option to hide treeview by default (there is even a package for that https://github.com/olmokramer/atom-autohide-tree-view ..)

**Comment by FalkF** [comment]
With the new IDE features came the outline card and it has the same problem.

**Comment by crazymykl** [comment]
If this is added, I hope it's optional rather than mandatory.

**Comment by chadhamre** [comment]
I want this too
